lib/tests/formulae.rb

```diff
@@ -82,39 +103,27 @@ def detect_formulae!(args:)
             "#{@argument} is not detected from GitHub Actions or a formula name!"
         end
 
-        # Use GitHub Actions variables for pull request jobs.
-        if ENV["GITHUB_BASE_REF"].present? && ENV["GITHUB_SHA"].present?
-          diff_start_sha1 = rev_parse("origin/#{ENV["GITHUB_BASE_REF"]}")
-          if diff_start_sha1.blank?
-            test git, "-C", repository, "fetch", "--depth=1",
-                 "origin", "+refs/heads/#{ENV["GITHUB_BASE_REF"]}",
-                 args: args
-            diff_start_sha1 = rev_parse("origin/#{ENV["GITHUB_BASE_REF"]}")
+        if ENV["GITHUB_REPOSITORY"].present?
+          # We're using GitHub Actions. Check the repo matches the tap we are using.
+          if tap&.full_name&.downcase == ENV["GITHUB_REPOSITORY"].downcase
+            diff_start_sha1, diff_end_sha1 = github_actions_revisions
           end
-          diff_end_sha1 = ENV["GITHUB_SHA"]
-        # Use GitHub Actions variables for branch jobs.
-        elsif ENV["GITHUB_SHA"].present?
-          diff_end_sha1 = diff_start_sha1 = ENV["GITHUB_SHA"]
-        # Otherwise just use the current SHA-1 (which may be overriden later)
-        else
-          if ENV["CI"] || ENV["GITHUB_ACTIONS"]
-            onoe <<~EOS
-              No known CI provider detected! If you are using GitHub Actions then we cannot find the expected  environment variables! Check you have e.g. exported them to a Docker container.
-            EOS
-          end
-          diff_end_sha1 = diff_start_sha1 = current_sha1
+        elsif ENV["CI"]
```

Comment:
GitHub Actions always sets this but it made not be passed through to e.g. Docker correctly. This is potentially an issue with `GITHUB_REPOSITORY` too so we may want to fail harder if we find _some_ GitHub Actions variables but not others (because the failure case is otherwise a warning and us building the wrong stuff...)

Reply:
So `ENV["CI"] || ENV["GITHUB_ACTIONS"]` like before or?

Reply:
Yeh, alternatively have a separate `ENV["GITHUB_ACTIONS"]` case which does a `raise` or `odie` because if we can't find the right environment variables: things are 💥